### PR TITLE
checker: fix smartcast

### PR DESCRIPTION
**Pull request by danieldaeschle**
fixes a bug where the smartcast wasn't done if the field is declared with mut but the root ident where the selector was made from isn't declared with mut.

### CODE REVIEW

File: vlib/v/checker/tests/sum_type_mutable_cast_err.out [hunk 1]
```diff
@@ -4,18 +4,11 @@ vlib/v/checker/tests/sum_type_mutable_cast_err.vv:23:10: error: cannot use opera
...
    24 |     }
-   25 |     f := Foo{Bar{Abc(0)}}
-vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27:14: error: cannot use operator `+` with `Abc`
-   25 |     f := Foo{Bar{Abc(0)}}
-   26 |     if f.b.a is int {
+   25 |     mut f2 := Foo2{Bar2{Abc(0)}}
+vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27:8: error: undefined ident: `f`
+   25 |     mut f2 := Foo2{Bar2{Abc(0)}}
+   26 |     if f2.b.a is int {
    27 |         _ := f.b.a + 5
```
Comment (dumblob) [inline vlib/v/checker/tests/sum_type_mutable_cast_err.out:11]: Shouldn't this be `_ := f2.b.a + 5`?

Comment (danieldaeschle) [reply vlib/v/checker/tests/sum_type_mutable_cast_err.out:11]: oh, yes!

File: vlib/v/checker/tests/sum_type_mutable_cast_err.vv [hunk 2]
```diff
@@ -22,10 +22,6 @@ fn main() {
 	if x is int {
 		_ := x + 5
 	}
-	f := Foo{Bar{Abc(0)}}
-	if f.b.a is int {
-		_ := f.b.a + 5
-	}
 	mut f2 := Foo2{Bar2{Abc(0)}}
 	if f2.b.a is int {
 		_ := f.b.a + 5
```
Comment (dumblob) [inline vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27]: Same as above.

Comment (danieldaeschle) [reply vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27]: Will fix it

Comment (danieldaeschle) [reply vlib/v/checker/tests/sum_type_mutable_cast_err.vv:27]: thanks

**Pull request merged**